File: simulations/scripts/zipf.R
#!/usr/bin/env Rscript
library(ggplot2)
library(RSvgDevice)

input=commandArgs(TRUE)[1]
output=commandArgs(TRUE)[2]

t <- read.table(paste(input, 'popularity-c1', sep=''))
devSVG(paste(output, 'zipf-c1.svg', sep=''))
ggplot(t, aes(seq_along(V1), V1)) + geom_line() + scale_x_log10() + scale_y_log10() + annotation_logticks()
dev.off()

# Why is this not working for alpha
model.lm <- lm(V1 ~ seq_along(V1), data = t)
model.lm

File: simulations/scripts/zipf.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys

def zipf(alpha, catalog_size):
    pi = [0] * (catalog_size+1)
    c = 1
    s = 0
    for i in range(1, catalog_size+1):
        pi[i] = c / i**alpha
        s += pi[i]
    for i in range(1, catalog_size+1):
        print i, pi[i]/s

    # 1/n^a    
if __name__ == '__main__':
    zipf(float(sys.argv[1]), int(sys.argv[2]))
